src/components/ProfileValidation/ReturnToAppButton.tsx

```diff
@@ -1,27 +1,106 @@
+import {
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  IconButton,
+} from '@mui/material'
+import CloseIcon from '@mui/icons-material/Close'
 import { Button } from '@mui/material'
 import { useAppContext } from 'AppContext'
 import React from 'react'
+import { Typography } from 'synapse-react-client'
+import theme from 'style/theme'
+import { NavLink } from 'react-router-dom'
 
 export const ReturnToAppButton: React.FC<{ children?: React.ReactNode }> = ({
   children,
 }) => {
   const { redirectURL } = useAppContext()
+  const [open, setOpen] = React.useState(false)
+  const onClose = () => setOpen(false)
   const redirectFn = () => {
     redirectURL
       ? window.location.assign(redirectURL)
       : window.location.assign('/authenticated/myaccount')
   }
-  const element = (
-    <Button variant="text" onClick={redirectFn} fullWidth>
-      {' '}
-      Cancel
+  const element = !children ? (
+    <Button variant="text" onClick={() => setOpen(true)} fullWidth>
+      Cancel Validation and Return to Website
     </Button>
-  )
-  if (!children) {
-    return element
-  } else {
-    return React.cloneElement(children as React.ReactElement, {
-      onClick: redirectFn,
+  ) : (
+    React.cloneElement(children as React.ReactElement, {
+      onClick: () => setOpen(true),
     })
-  }
+  )
+
+  return (
+    <>
```

Comment:
if we will need to use this type of dialog it makes sense to create it as a shared control. For now it's a combination of inline styles and theme